test/integration/propControls/index.spec.ts

```diff
@@ -40,35 +24,28 @@ test('can control component prop values in properties control panel', async ({
   const firstInputLocator = canvasInputLocator.first();
   await clickCenter(page, firstInputLocator);
 
-  await editorModel.componentEditor.waitFor();
+  await editorModel.componentEditor.waitFor({ state: 'visible' });
+
+  const labelControlInput = editorModel.componentEditor.getByLabel('label', { exact: true });
 
-  const labelControlInputValue = await editorModel.componentEditor
-    .locator(`label:text-is("label")`)
-    .inputValue();
+  const labelControlInputValue = await labelControlInput.inputValue();
 
   expect(labelControlInputValue).toBe('textField1');
 
   // Change component prop values directly
-
   const TEST_VALUE_1 = 'value1';
-
-  const getValueControlInputValue = async () =>
-    editorModel.componentEditor.locator(`label:text-is("value")`).inputValue();
-
-  expect(await getValueControlInputValue()).not.toBe(TEST_VALUE_1);
+  const valueControl = editorModel.componentEditor.getByLabel('value', { exact: true });
+  expect(await valueControl.inputValue()).not.toBe(TEST_VALUE_1);
   await firstInputLocator.fill(TEST_VALUE_1);
-  expect(await getValueControlInputValue()).toBe(TEST_VALUE_1);
+  expect(await valueControl.inputValue()).toBe(TEST_VALUE_1);
 
   // Change component prop values through controls
-
-  const firstInputLabelLocator = await getInputElementLabelLocator(editorModel, firstInputLocator);
   const TEST_VALUE_2 = 'value2';
+  const inputByLabel = editorModel.appCanvas.getByLabel(TEST_VALUE_2, { exact: true });
+  await expect(inputByLabel).toHaveCount(0);
+  await labelControlInput.click();
+  await labelControlInput.fill('');
+  await labelControlInput.fill(TEST_VALUE_2);
 
-  await expect(firstInputLabelLocator).not.toHaveText(TEST_VALUE_2);
-
-  const labelControlInputLocator = await getPropControlInputLocator(editorModel, 'label');
-  await labelControlInputLocator.fill('');
-  await labelControlInputLocator.fill(TEST_VALUE_2);
-
-  await expect(firstInputLabelLocator).toHaveText(TEST_VALUE_2);
+  await inputByLabel.waitFor({ state: 'visible' });
```

Comment:
So we're not checking if the value in the label matches in the end anymore? Was there any issue with that?

Reply:
Yes we are, if the value doesn't match this selector wouldn't locate elements

Reply:
oh it checks with that value already, i missed that. looks good then!